setup.py

```diff
@@ -1,91 +1,81 @@
-if '--debug' in sys.argv or '-g' in sys.argv or 'CPPDEBUG' in os.environ:
-    mm_compile_args['msvc'].append('/DCPPDEBUG')
-    base_compile_args['unix'] = ['-std=c++17', '-Wall', '-O0', '-g', '-fipa-pure-const']
-
-
-glasgow_compile_args = {
-    'msvc': base_compile_args['msvc'] + ['/external:W4', '/external:I external', '/DUSE_PORTABLE_SNIPPETS_BUILTIN'],
-    'unix': base_compile_args['unix'] + ['-isystemexternal', '-DUSE_PORTABLE_SNIPPETS_BUILTIN']
-}
-
-if 'SAFE_COORDS' in os.environ:
-    mm_compile_args['msvc'].append('/DSAFE_COORDS')
-    mm_compile_args['unix'].append('-DSAFE_COORDS')
-
-extra_compile_args = {
-    'mm': mm_compile_args,
-    'glasgow': glasgow_compile_args,
-}
-
-extra_link_args = {
-    'msvc': [],
-    'unix': ['-std=c++17'],
-}
-
-class build_ext_compiler_check(build_ext):
-    def build_extensions(self):
-        compiler = self.compiler.compiler_type
-
-        for ext in self.extensions:
-            arg_key = ext.extra_compile_args
-            if arg_key:
-                ext.extra_compile_args = extra_compile_args[arg_key][compiler]
-
-        link_args = extra_link_args[compiler]
-        for ext in self.extensions:
-            ext.extra_link_args = link_args
-
-        build_ext.build_extensions(self)
-        build_ext.build_extensions(self)
-
-
-class Extension(extension.Extension, object):
-    pass
-
-
-ext = '.pyx' if USE_CYTHON else '.cpp'
-ext_c = '.pyx' if USE_CYTHON else '.c'
-
+DEBUG = '--debug' in sys.argv or '-g' in sys.argv or 'CPPDEBUG' in os.environ
+SAFE_COORDS = 'SAFE_COORDS' in os.environ
+
+extra_compile_args = []
+extra_compile_args_glasgow = []  # specific to glasgow
+extra_compile_args_minorminer = []  # specific to minorminer
+if platform.system().lower() == "windows":
+    extra_compile_args.extend([
+        '/std:c++17',
+        '/MT',
+        '/EHsc',
+        ])
+
+    extra_compile_args_glasgow.extend([
+        '/external:W4',
+        '/external:I external',
+        '/DUSE_PORTABLE_SNIPPETS_BUILTIN',
+        ])
+
+    extra_compile_args_minorminer.extend([
+        '/DCPPDEBUG' if DEBUG else None,
+        '/DSAFE_COORDS' if SAFE_COORDS else None,
+        ])
+
+else:  # Unix
+    extra_compile_args.extend([
+        '-std=c++17',
+        '-Wall',
+        '-Wno-format-security',
+        '--O0' if DEBUG else '-Ofast',
+        '-fomit-frame-pointer',
+        '-fipa-pure-const' if DEBUG else None,
+        '-g' if DEBUG else '-g1',
+        ])
+
+    extra_compile_args_glasgow.extend([
+        '-isystemexternal',
+        '-DUSE_PORTABLE_SNIPPETS_BUILTIN',
+        ])
+
+    extra_compile_args_minorminer.extend([
+        '-fno-rtti',
+        '-DSAFE_COORDS' if SAFE_COORDS else None,
+        ])
+
+# fitler out any None or empty arguments
+extra_compile_args = [arg for arg in extra_compile_args if arg]
+extra_compile_args_glasgow = [arg for arg in extra_compile_args_glasgow if arg]
+extra_compile_args_minorminer = [arg for arg in extra_compile_args_minorminer if arg]
```

Comment:
Minor, but could use `filter(None, extra_compiler_args)` here instead.

Reply:
Ah, I didn't know about `None` as the first arg to filter. Neat!